src/query/functions-v2/src/scalars/hash.rs

```diff
@@ -0,0 +1,309 @@
+use common_expression::FunctionRegistry;
+use common_expression::Scalar;
+use md5::Digest;
+use md5::Md5 as Md5Hasher;
+use naive_cityhash::cityhash64_with_seed;
+use num_traits::AsPrimitive;
+
+use crate::scalars::string::vectorize_string_to_string;
+
+pub fn register(registry: &mut FunctionRegistry) {
+    registry.register_passthrough_nullable_1_arg::<StringType, StringType, _, _>(
+        "md5",
+        FunctionProperty::default(),
+        |_| FunctionDomain::MayThrow,
+        vectorize_string_to_string(
+            |col| col.data.len() * 32,
+            |val, output, _| {
+                // TODO md5 lib doesn't allow encode into buffer...
+                let old_len = output.data.len();
+                output.data.resize(old_len + 32, 0);
+                hex::encode_to_slice(
+                    Md5Hasher::digest(val).as_slice(),
+                    &mut output.data[old_len..],
+                )
+                .map_err(|err| format!("unable to hash with md5: {err}"))?;
+                output.commit_row();
+                Ok(())
+            },
+        ),
+    );
+
+    registry.register_passthrough_nullable_1_arg::<StringType, StringType, _, _>(
+        "sha",
+        FunctionProperty::default(),
+        |_| FunctionDomain::MayThrow,
+        vectorize_string_to_string(
+            |col| col.data.len() * 40,
+            |val, output, _| {
+                let old_len = output.data.len();
+                output.data.resize(old_len + 40, 0);
+                // TODO sha1 lib doesn't allow encode into buffer...
+                let mut m = ::sha1::Sha1::new();
+                sha1::digest::Update::update(&mut m, val);
+                hex::encode_to_slice(m.finalize().as_slice(), &mut output.data[old_len..])
+                    .map_err(|err| format!("unable to hash with sha: {err}"))?;
+                output.commit_row();
+                Ok(())
+            },
+        ),
+    );
+    registry.register_aliases("sha", &["sha1"]);
+
+    registry.register_passthrough_nullable_1_arg::<StringType, StringType, _, _>(
+        "blake3",
+        FunctionProperty::default(),
+        |_| FunctionDomain::MayThrow,
+        vectorize_string_to_string(
+            |col| col.data.len() * 64,
+            |val, output, _| {
+                let old_len = output.data.len();
+                output.data.resize(old_len + 64, 0);
+                // TODO blake3 lib doesn't allow encode into buffer...
+                hex::encode_to_slice(blake3::hash(val).as_bytes(), &mut output.data[old_len..])
+                    .map_err(|err| format!("unable to hash with blake3: {err}"))?;
+                output.commit_row();
+                Ok(())
+            },
+        ),
+    );
+
+    for ty in MAX_NUMERICS_TYPES {
+        with_number_mapped_type!(|NUM_TYPE| match ty {
+            NumberDataType::NUM_TYPE => {
+                registry.register_passthrough_nullable_2_arg::<StringType, NumberType<NUM_TYPE>, StringType, _, _>(
```

Comment:
`<StringType, Int64Type, StringType>` is enough here. 

The second arg is mostly constant.

Reply:
I do a test, and we support this in function v1...

```sql
select sha2('x',224.0);
+----------------------------------------------------------+
| sha2('x', 224)                                           |
+----------------------------------------------------------+
| 54a2f7f92a5f975d8096af77a126edda7da60c5aa872ef1b871701ae |
+----------------------------------------------------------+

```